src/imitation_loss/imitation_loss.jl

```diff
@@ -0,0 +1,76 @@
+"""
+    ImitationLoss{L,R,P}
+
+Generic imitation loss: max_y ℓ(y, y̅) + α θᵀ(y - y̅) - (Ω(y) - Ω(y̅)).
+
+When `ℓ = 0`, this loss is equivalent to a [`FenchelYoungLoss`](@ref).
+When `Ω = 0`, this loss is equivalent to the [`StructuredSVMLoss`](@ref).
+
+# Fields
+- `maximizer::P`: function that computes
+    argmax_y ℓ(y, y̅) + α θᵀ(y - y̅) - (Ω(y) - Ω(y̅)), takes (θ, y_true, kwargs...)
+    or (θ, t_true, kwargs...) as input
+- `ℓ::L`: base loss, can either take (y, y_true) or (y, t_true) as input
+- `Ω::R`: regularization, takes y as input
+- `α::Float64`: default value of 1.0
+"""
+struct ImitationLoss{P,L,R}
+    maximizer::P
+    ℓ::L
+    Ω::R
+    α::Float64
+end
+
+function Base.show(io::IO, l::ImitationLoss)
+    (; ℓ, Ω, maximizer, α) = l
+    return print(io, "ImitationLoss($maximizer, $ℓ, $Ω, $α)")
+end
+
+"""
+    ImitationLoss(maximizer[; ℓ=(y,t_true)->0.0, Ω=y->0.0, α=1.0])
+
+Shorter constructor with defaults.
+"""
+function ImitationLoss(
+    maximizer; ℓ=(y, t_true) -> 0.0, base_loss=nothing, Ω=y -> 0.0, omega=nothing, α=1.0
+)
+    if isnothing(base_loss) && isnothing(omega)
+        return ImitationLoss(maximizer, ℓ, Ω, α)
+    end
+    if isnothing(base_loss)
+        return ImitationLoss(maximizer, ℓ, omega, α)
+    end
+    if isnothing(omega)
+        return ImitationLoss(maximizer, base_loss, Ω, α)
+    end
+    return ImitationLoss(maximizer, base_loss, omega, α)
+end
+
+getytrue(t_true) = error("not implemented")  # TODO: find a better function name ?
+getytrue(t_true::NamedTuple) = t_true.y_true
```

Comment:
Maybe `t_true.y` is enough?

Reply:
It might be enough, but `y_true` might avoid confusions when defining the base losses :
```julia
function spo_base_loss(y, t_true)
    (; θ_true, y_true) = t_true
    return dot(θ_true, y_true) - dot(θ_true, y)
end
```

Reply:
better safe than sorry then